File: firebase/functions/lib/events/live_meetings/breakouts/check_assign_to_breakouts_server.dart
import 'dart:async';

import 'package:firebase_functions_interop/firebase_functions_interop.dart';
import 'check_assign_to_breakouts.dart';
import '../../../on_request_method.dart';
import 'package:data_models/cloud_functions/requests.dart';

/// This is a wrapper on our [CheckAssignToBreakouts] function.
/// Whereas that one is called from the client, this one is called from the
/// server at a scheduled time depending on when the event is scheduled for.
class CheckAssignToBreakoutsServer
    extends OnRequestMethod<CheckAssignToBreakoutsRequest> {
  CheckAssignToBreakoutsServer()
      : super(
          'CheckAssignToBreakoutsServer',
          (jsonMap) => CheckAssignToBreakoutsRequest.fromJson(jsonMap),
          runWithOptions: RuntimeOptions(
            timeoutSeconds: 240,
            memory: '4GB',
            minInstances: 0,
          ),
        );

  @override
  Future<String> action(CheckAssignToBreakoutsRequest request) async {
    await CheckAssignToBreakouts().checkAssignToBreakouts(request, 'server');

    return '';
  }
}

File: firebase/functions/src/events/live_meetings/breakouts/check_assign_to_breakouts_server.ts
import { OnRequestMethod } from '../../../on_request_method';
import { CheckAssignToBreakouts } from './check_assign_to_breakouts';

interface CheckAssignToBreakoutsRequest {
  eventPath: string;
  breakoutSessionId: string;
}

export class CheckAssignToBreakoutsServer extends OnRequestMethod<CheckAssignToBreakoutsRequest> {
  constructor() {
    super(
      'CheckAssignToBreakoutsServer',
      (jsonMap) => jsonMap as CheckAssignToBreakoutsRequest,
      { runWithOptions: { timeoutSeconds: 240, memory: '4GB', minInstances: 0 } }
    );
  }

  async action(request: CheckAssignToBreakoutsRequest): Promise<string> {
    await new CheckAssignToBreakouts().checkAssignToBreakouts(request, 'server');
    return '';
  }
}
